server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -715,11 +716,22 @@ public Pair<List<? extends VpcOffering>, Integer> listVpcOfferings(ListVPCOfferi
         final String state = cmd.getState();
         final Long startIndex = cmd.getStartIndex();
         final Long pageSizeVal = cmd.getPageSizeVal();
+        final Long domainId = cmd.getDomainId();
         final Long zoneId = cmd.getZoneId();
         final Filter searchFilter = new Filter(VpcOfferingJoinVO.class, "sortKey", QueryService.SortKeyAscending.value(), null, null);
         searchFilter.addOrderBy(VpcOfferingJoinVO.class, "id", true);
         final SearchCriteria<VpcOfferingJoinVO> sc = vpcOfferingJoinDao.createSearchCriteria();
 
+        if (domainId != null) {
+            Domain domain = _entityMgr.findById(Domain.class, domainId);
+            if (domain == null) {
+                throw new InvalidParameterValueException("Unable to find the domain by id=" + domainId);
+            }
+            if (!domainDao.isChildDomain(caller.getDomainId(), domainId)) {
+                throw new InvalidParameterValueException(String.format("Unable to list VPC offerings for domain: %s as caller does not have access for it", domain.getUuid()));
+            }
+        }
```

Comment:
can this go in a separate `verifyDomainId()`, please?

Reply:
Done